build error in default_configuration(): label can only be part of a statement (configuration.c:99)

**Which version do you want to build? Please name version and source of the tarball/package.**

git master (HEAD), pulled directly from `RRZE-HPC/likwid` master. I Encountered this while building likwid as part of `TheHPXProject/hpx`'s CI-CD run on a PR . Confirmed still present in current master, `src/configuration.c`, `default_configuration()`, line 99.

**Which architecture do you want to build for?**

x86_64

**To Reproduce**

No changes to `config.mk`. Default GCC build via the Makefile (`make` in `src/`), `LIKWID_USE_PERFEVENT` not defined, so the `use_hardcoded:` goto label path in `default_configuration()` is taken.

Build environment: `stellargroup/build_env:17` (Ubuntu-based CI docker image used by the HPX project), exact GCC version not confirmed

**Output log**

```
GCC/configuration.c: In function 'default_configuration':
GCC/configuration.c:99:5: error: a label can only be part of a statement and a declaration is not a statement
   99 |     const char *hardcodedAccessDaemonPath = TOSTRING(ACCESSDAEMON);
      |     ^~~~~
make: *** [Makefile:369: GCC/configuration.o] Error 1
```

<img width="1385" height="600" alt="Image" src="https://github.com/user-attachments/assets/2cbd0ee7-9e1d-478e-aa7a-a7c11a2618f4" />

The issue is that `use_hardcoded:` is immediately followed by a declaration with no statement in between, which is invalid in C until C23 (a label must be attached to a statement, and a declaration is not a statement). Fix would be to insert an empty statement after the label  e.g.:

```c
#ifndef LIKWID_USE_PERFEVENT
use_hardcoded:
    ;
#endif
    const char *hardcodedAccessDaemonPath = TOSTRING(ACCESSDAEMON);
```

happy to send a PR with this fix if useful

--- a/src/configuration.c
+++ b/src/configuration.c
@@ -94,6 +94,7 @@ static int default_configuration(void)
 
 #ifndef LIKWID_USE_PERFEVENT
 use_hardcoded:
+    ;
 #endif
 
     const char *hardcodedAccessDaemonPath = TOSTRING(ACCESSDAEMON);

